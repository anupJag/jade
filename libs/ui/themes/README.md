# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
