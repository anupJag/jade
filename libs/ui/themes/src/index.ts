export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
