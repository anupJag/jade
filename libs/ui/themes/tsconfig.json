{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
