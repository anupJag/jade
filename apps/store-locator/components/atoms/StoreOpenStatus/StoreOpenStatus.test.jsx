import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { ThemeProvider } from '@jade/ui/themes';

import { StoreOpenStatus } from './index';

describe('StoreOpenStatus', () => {
  let StoreOpenStatusComponent = '';

  beforeEach(() => {
    const storeTiming = {
      mon: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      tue: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      wed: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      thu: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      fri: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      sat: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
      sun: {
        openingTime: '07:00',
        closingTime: '23:00',
      },
    };
    const offset = Number('+10');

    StoreOpenStatusComponent = render(
      <ThemeProvider>
        <StoreOpenStatus storeTiming={storeTiming} offset={offset} />
      </ThemeProvider>,
    );
  });

  it('should render store open status correctly', async () => {
    const { asFragment } = StoreOpenStatusComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});
