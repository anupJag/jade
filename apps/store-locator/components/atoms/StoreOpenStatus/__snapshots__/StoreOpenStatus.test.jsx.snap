// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreOpenStatus should render store open status correctly 1`] = `
<DocumentFragment>
  <svg
    class="css-kbr51z"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 384 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0z"
    />
  </svg>
</DocumentFragment>
`;
